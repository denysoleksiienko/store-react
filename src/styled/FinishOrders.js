import styled from 'styled-components';

export const OrderWrap = styled.div`
  margin-top: 50px;
  margin-left: 16px;
`;
