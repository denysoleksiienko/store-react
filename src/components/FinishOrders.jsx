import Col from 'react-bootstrap/Col';
import { Title, OrderWrap } from 'styled';

export const FinishOrders = ({ user }) => (
  <Col lg={7}>
    <Title>Thank you for your order!</Title>
    <OrderWrap>
      <p>
        Order number is: <b>188787788</b>
      </p>
      <p>
        You will recieve an email confirmation shortly to <b>{user.email}</b>
      </p>
      <p>Estimated delivery Day is Friday 1st April 2016</p>
      <p>Print Recipe</p>
    </OrderWrap>
  </Col>
);
