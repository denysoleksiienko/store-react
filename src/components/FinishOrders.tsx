import React from 'react';
import Col from 'react-bootstrap/Col';

import { IFinishOrder } from 'interfaces/Orders';

import { Title, Wrap, Inner, Button, SpanEmail, SpanDelivery, OrderNumber } from 'styled';

export const FinishOrders: React.FC<IFinishOrder> = ({ user }) => (
  <Col lg={7}>
    <Wrap>
      <Title>Thank you for your order!</Title>
      <Inner>
        <OrderNumber>
          Order number is: <b>188787788</b>
        </OrderNumber>
        <p>
          You will recieve an email confirmation shortly to <SpanEmail>{user.email}</SpanEmail>
        </p>
        <p>
          Estimated delivery Day is <br />
          <SpanDelivery>Friday 1st April 2016</SpanDelivery>
        </p>
        <Button onClick={() => window.print()}> Print Recipe </Button>
      </Inner>
    </Wrap>
  </Col>
);
