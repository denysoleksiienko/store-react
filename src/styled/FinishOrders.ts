import styled from 'styled-components';
import { COLORS } from 'constants/colors';

export const Wrap = styled.div`
  margin-top: 50px;
  font-size: 18px;
  @media print {
    color: #000;
  }
`;

export const Inner = styled.div`
  margin-top: 50px;
  margin-left: 16px;
`;

export const OrderNumber = styled.p`
  font-size: 20px;
`;

export const SpanEmail = styled.span`
  font-weight: bold;
  color: ${COLORS.orchid};
  text-decoration: underline;
`;

export const SpanDelivery = styled.span`
  font-weight: bold;
  font-size: 20px;
  line-height: 36px;
`;

export const Button = styled.button`
  padding: 0;
  font-weight: bold;
  color: ${COLORS.orchid};
  border: none;
  text-decoration: underline;
  background-color: #fff;
  &:active,
  :focus,
  :hover {
    outline: none;
  }
`;
