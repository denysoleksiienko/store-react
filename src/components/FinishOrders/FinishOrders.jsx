import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';

import { RowWrap, Title, OrderWrap, ColOrders } from 'styled';

import { Orders } from 'components';

export const FinishOrders = () => (
  <Container>
    <RowWrap>
      <Col lg={7}>
        <Title>Thank you for your order!</Title>
        <OrderWrap>
          <p>Order number is: 188787788</p>
          <p>You will recieve an email confirmation shortly to janathan.smith@gmail.com</p>
          <p>Estimated delivery Day is Friday 1st April 2016</p>
          <p>Print Recipe</p>
        </OrderWrap>
      </Col>
      <ColOrders lg={5}>
        <Orders />
      </ColOrders>
    </RowWrap>
  </Container>
);
