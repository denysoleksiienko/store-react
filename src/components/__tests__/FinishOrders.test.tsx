import renderer from 'react-test-renderer';

import { FinishOrders } from '../FinishOrders';

const props = {
  user: {
    email: 'google@gmail.com',
  },
};

describe('FinishOrders component', () => {
  it('should match snapshot', () => {
    const wrapper = renderer.create(<FinishOrders {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
