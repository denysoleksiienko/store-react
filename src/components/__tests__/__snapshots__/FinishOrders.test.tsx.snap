// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishOrders component should match snapshot 1`] = `
<div
  className="col-lg-7"
>
  <div
    className="sc-jrAGrp gvvVlL"
  >
    <h1
      className="sc-dmlrTW eDBzFv"
    >
      Thank you for your order!
    </h1>
    <div
      className="sc-kEjbxe gwSbRr"
    >
      <p
        className="sc-iqHYGH lkcwNC"
      >
        Order number is: 
        <b>
          188787788
        </b>
      </p>
      <p>
        You will recieve an email confirmation shortly to 
        <span
          className="sc-crrsfI AcZYB"
        >
          google@gmail.com
        </span>
      </p>
      <p>
        Estimated delivery Day is 
        <br />
        <span
          className="sc-dQppl cwhHZX"
        >
          Friday 1st April 2016
        </span>
      </p>
      <button
        className="sc-bqyKva lhnwhK"
        onClick={[Function]}
      >
         Print Recipe 
      </button>
    </div>
  </div>
</div>
`;
